content/blog/build-and-release-hugo-blog-site-using-azure-pipelines.md

```diff
@@ -0,0 +1,208 @@
+
+![Initial repository setup](../../static/images/build-and-release-hugo-site-using-azure-pipelines/initial-repo-setup.png)
+<!-- ![Initial repository setup](../../../../../images/build-and-release-hugo-site-using-azure-pipelines/initial-repo-setup.png) -->
+
+As per the instructions I used a `deploy.sh` script to generate the Hugo static website in the public folder and push any changes to `ronaldbosma.github.io`. This has a few disadvantages though.
+
+First off it's a manual action which can be forgotten easily. Also, I'm a developer and I dislike doing things manually.
+
+I also prefer to create separate branches for my posts so I can create a pull request and ask my peers for feedback. The `deploy.sh` script can be run from every branch. Possibly publishing something I don't want to or removing changes I haven't merged from master yet.
+
+The last problem I had was that for some reason the link to the submodule seemed to disappear. Making pushing any changes to `ronaldbosma.github.io` impossible. As a pragmatic solution I just removed and added the submodule manually. Which was cumbersome.
+
+What I wanted was to automatically build and release my blog whenever I push a change in the master branch of my `blog` repository. I used Azure Pipelines to make this happen. The image below shows the various steps that are executed in the pipeline.
+
+![Pipeline](../../static/images/build-and-release-hugo-site-using-azure-pipelines/pipeline.png)
+<!-- ![Pipeline](../../../../../images/build-and-release-hugo-site-using-azure-pipelines/pipeline.png) -->
+
+First a build pipeline is triggered whenever I push to master. Using a Hugo build task the static site is generated and published as an artifact of the build. When the build succeeds a release pipeline is triggered which will take the site and push this into the `ronaldbosma.github.io` repository using a Publish to GitHub Pages task.
+
+Although I'm interacting with a git repository when I publish my site, I decided to put this step in a release pipeline and not a build pipeline. I'm releasing my site here. Not building it. This also allows me to add more stages to for example a testing environment or change the target location and type of my site altogether without affecting my build pipeline.
+
+There's another advantage to this solution besides having to do less manual work when publishing new changes to my blog. I can set a publish date in my posts. Posts with a publish date in the future will be ignored when generating my Hugo site. By scheduling my build pipeline to trigger at a regular interval I'm able to schedule posts to be published in the future.
+
+In the next section of this post you'll find a step-by-step explanation of what I did to create this pipeline.
+
+- [Step 1: Prerequisites](#step-1-prerequisites)
```

Comment:
De stappen zonder punt zijn goed 'geanchored' naar hun bijbehorende kopjes, maar de de sub punten linken door naar bijvoobeeld 1-1-install-the-hugo-extension ipv van 11-install-the hugo extension. Als je in de url kijkt nadat je ze klikt zie je wat ik bedoel.

Reply:
Hugo werkt met -, maar GitHub blijkbaar zonder. Zo doen ze het iig op de site.